bbq/compiler/compiler.go

```diff
@@ -2991,12 +2991,21 @@ func (c *Compiler[_, _]) compileInitializer(declaration *ast.SpecialFunctionDecl
 	}
 
 	if address == common.ZeroAddress {
-		c.emit(
-			opcode.InstructionNewComposite{
-				Kind: kind,
-				Type: typeIndex,
-			},
-		)
+		if typeName == commons.TransactionWrapperCompositeName {
```

Comment:
Might be good to also check the location, as also user-defined composites might be named `transaction`. Maybe add a test that e.g. `struct transaction {}` isn't miscompiled

Reply:
`transaction` is a keyword, so `struct transaction {}` isn't allowed by the parser.

Reply:
Added test: https://github.com/onflow/cadence/pull/4116/commits/f9ad815860cabfb687fde766f2f866ee152a98f8

Reply:
Nice! Added the additional check for the location in 1909badce3aaae4b26a96d6f9faead4e9d283170, just in case